Cython/Utility/Embed.c

```diff
@@ -23,18 +23,49 @@ static int __Pyx_main(int argc, wchar_t **argv)
     m = fpgetmask();
     fpsetmask(m & ~FP_X_OFL);
 #endif
+#if PY_VERSION_HEX < 0x03080000
     if (argc && argv)
         Py_SetProgramName(argv[0]);
+#endif
 
     #if PY_MAJOR_VERSION < 3
     if (PyImport_AppendInittab("%(module_name)s", init%(module_name)s) < 0) return 1;
     #else
     if (PyImport_AppendInittab("%(module_name)s", PyInit_%(module_name)s) < 0) return 1;
     #endif
 
+#if PY_VERSION_HEX < 0x03080000
     Py_Initialize();
     if (argc && argv)
         PySys_SetArgv(argc, argv);
+#else
+    PyStatus status;
+
+    PyConfig config;
```

Comment:
We still generate C89, so these declarations need to go to the top. Or, use a nested block for this, that's probably better for scoping reasons.

Reply:
Done